policy/assertion.go

```diff
@@ -56,3 +65,24 @@ func NewAssertion(action, resource, effect string) (*Assertion, error) {
 		}(),
 	}, nil
 }
+
+// AssertionJSON represents json format of Assertion
+type AssertionJSON struct {
```

Comment:
It shouldn't be public. Or it should be inline object for temp use in MarshalJSON function.
```suggestion
type assertionJSON struct {
```

Reply:
It is the JSON definition. There is nothing to hide.
And user need this struct to implement the decode function.

Reply:
It makes user confuse which object should be use, also it is internal use object for only marshaling use.

Reply:
Also I think this object should be 
```suggestion
type assertionJSON struct {
Assertion
RegStr string
```

Reply:
reverted